otlp/translator.go

```diff
@@ -0,0 +1,461 @@
+// Copyright 2020 Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otlp
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/jaegertracing/jaeger/model"
+
+	splunksapm "github.com/signalfx/sapm-proto/gen"
+	otlpcoltrace "github.com/signalfx/sapm-proto/gen/otlp/collector/trace/v1"
+	otlpcommon "github.com/signalfx/sapm-proto/gen/otlp/common/v1"
+	otlpresource "github.com/signalfx/sapm-proto/gen/otlp/resource/v1"
+	otlptrace "github.com/signalfx/sapm-proto/gen/otlp/trace/v1"
+)
+
+// otlpToSAPM translates otlp trace proto into the SAPM Proto.
+func otlpToSAPM(td otlpcoltrace.ExportTraceServiceRequest) (*splunksapm.PostSpansRequest, error) {
```

Comment:
Assuming this is copied from Otel code, make sure to keep Otel copyright at the top of this file.

Reply:
Can you give me an example?

Reply:
Start the files with:
```
// Copyright 2020 Splunk, Inc.
// Copyright The OpenTelemetry Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//       http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
```

Reply:
Done.